test/unit/math/rev/fun/to_var_value_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <stan/math/rev.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+TEST(AgradRevMatrix, to_var_value_matrix_test) {
+  Eigen::MatrixXd val(2, 3);
+  val << 1, 2, 3, 4, 5, 6;
+  Eigen::MatrixXd adj(2, 3);
+  val << 4, 5, 6, 7, 8, 9;
+  Eigen::Matrix<stan::math::var, Eigen::Dynamic, Eigen::Dynamic> var = val;
```

Comment:
[nbd] I'd make this var_mat instead of just `var`

Reply:
fixed